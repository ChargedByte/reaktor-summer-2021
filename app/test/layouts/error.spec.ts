import { createLocalVue, mount, RouterLinkStub } from '@vue/test-utils'

import Vuetify from 'vuetify'

import ErrorPage from '~/layouts/error.vue'

describe('ErrorPage', () => {
  const localVue = createLocalVue()
  let vuetify: Vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  test('is a Vue instance', () => {
    const wrapper = mount(ErrorPage, {
      localVue,
      vuetify,
      stubs: { RouterLink: RouterLinkStub },
      propsData: { error: { statusCode: 500 } },
    })
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly: 404', () => {
    const wrapper = mount(ErrorPage, {
      localVue,
      vuetify,
      stubs: { RouterLink: RouterLinkStub },
      propsData: { error: { statusCode: 404 } },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('renders correctly: 500', () => {
    const wrapper = mount(ErrorPage, {
      localVue,
      vuetify,
      stubs: { RouterLink: RouterLinkStub },
      propsData: { error: { statusCode: 500 } },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
