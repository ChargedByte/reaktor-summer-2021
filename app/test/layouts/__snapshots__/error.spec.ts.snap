// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage renders correctly: 404 1`] = `
"<div class=\\"container container-error pa-0 container--fluid fill-height\\">
  <div class=\\"row align-center justify-center\\">
    <div class=\\"transparent v-card v-card--flat v-sheet theme--dark\\">
      <div class=\\"row align-center justify-center\\"><span class=\\"text-status\\">404</span></div>
      <div class=\\"row align-center justify-center\\"><span class=\\"text-message\\">Page Not Found</span></div>
      <div class=\\"row mt-10 align-center justify-center\\"><a class=\\"v-btn v-btn--is-elevated v-btn--has-bg v-btn--router theme--dark v-size--default\\"><span class=\\"v-btn__content\\">Go Home</span></a></div>
    </div>
  </div>
</div>"
`;

exports[`ErrorPage renders correctly: 500 1`] = `
"<div class=\\"container container-error pa-0 container--fluid fill-height\\">
  <div class=\\"row align-center justify-center\\">
    <div class=\\"transparent v-card v-card--flat v-sheet theme--dark\\">
      <div class=\\"row align-center justify-center\\"><span class=\\"text-status\\">500</span></div>
      <div class=\\"row align-center justify-center\\"><span class=\\"text-message\\">An Error Occurred</span></div>
      <div class=\\"row mt-10 align-center justify-center\\"><a class=\\"v-btn v-btn--is-elevated v-btn--has-bg v-btn--router theme--dark v-size--default\\"><span class=\\"v-btn__content\\">Go Home</span></a></div>
    </div>
  </div>
</div>"
`;
